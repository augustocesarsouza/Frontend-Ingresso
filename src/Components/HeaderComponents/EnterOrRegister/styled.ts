import styled from 'styled-components';

export const ContainerEnterOrRegisterMain = styled.div`
  position: absolute;
  left: -254px;
  top: 50px;
  background-color: #191919;
  width: 51rem;
  height: 26rem;
  display: flex;
  padding: 30px 10px;
  z-index: 9999;

  svg {
    position: relative;
    color: #191919;
    font-size: 39px;
    left: 249px;
    bottom: 52px;
  }
`

export const ContainerEnterOrRegister = styled.div`
  width: 50%;
  padding-right: 15px;
`

export const ContainerRegister = styled.div`
  display: flex;
  flex-direction: column;
  gap: 30px;
`

export const H2 = styled.h2`
  font-size: 17px;
  color: white;
  font-weight: 600;
`

export const P = styled.p`
  font-size: 14px;
  color: white;
`

export const ButtonRegister = styled.button`
  padding: 10px;
  border: none;
  background: #0099ff;
  color: #000000;
  font-size: 14px;
  font-weight: 700;
  border-radius: 6px;
  cursor: pointer;
`

export const ContainerLineWhite = styled.div`
  border-right: 1px solid #ffffff5e;
  margin-left: 13px;
`

export const ContainerLogin = styled.div`
  width: 50%;
  padding-left: 20px;
  display: flex;
  flex-direction: column;
  gap: 25px;
`

export const link = styled.a`
  color: #2196f3;
  font-size: 15px;
  cursor: pointer;
`