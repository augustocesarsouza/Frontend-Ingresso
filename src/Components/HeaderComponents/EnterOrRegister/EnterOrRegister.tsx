import { faCaretUp } from '@fortawesome/free-solid-svg-icons';
import * as Styled from './styled';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { useNavigate } from 'react-router-dom';

const EnterOrRegister = ({ enterOrRegister }) => {
  const nav = useNavigate();
  const handleRegisterAccount = () => {
    nav('/minha-conta/cadastro');
  };

  const handleLogin = () => {
    nav('/minha-conta/login');
  };

  return (
    <>
      {enterOrRegister && (
        <Styled.ContainerEnterOrRegisterMain>
          <FontAwesomeIcon icon={faCaretUp} />
          <Styled.ContainerEnterOrRegister>
            <Styled.ContainerRegister>
              <Styled.H2>Ainda não é cliente Ingresso.com?</Styled.H2>
              <Styled.P>
                Compre ingressos e combos de pipoca para ir ao cinema com segurança e mais
                comodidade!
              </Styled.P>
              <Styled.ButtonRegister onClick={handleRegisterAccount}>
                CRIAR UMA NOVA CONTA
              </Styled.ButtonRegister>
            </Styled.ContainerRegister>
          </Styled.ContainerEnterOrRegister>
          <Styled.ContainerLineWhite></Styled.ContainerLineWhite>
          <Styled.ContainerLogin>
            <Styled.H2>Cliente Ingresso.com?</Styled.H2>
            <Styled.link onClick={handleLogin}>Entrar na minha Conta</Styled.link>
          </Styled.ContainerLogin>
        </Styled.ContainerEnterOrRegisterMain>
      )}
    </>
  );
};

export default EnterOrRegister;
