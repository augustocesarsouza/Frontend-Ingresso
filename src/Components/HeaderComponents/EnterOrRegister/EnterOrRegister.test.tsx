import { render, screen } from '@testing-library/react';
import EnterOrRegister from './EnterOrRegister';
import { MemoryRouter } from 'react-router-dom';

describe('EnterOrRegister', () => {
  test('Should render svg triangle', () => {
    const { container } = render(
      <MemoryRouter>
        <EnterOrRegister enterOrRegister={true} />
      </MemoryRouter>
    );

    const svgEl = container.querySelector("[data-icon='caret-up']") as SVGAElement;
    expect(svgEl).toBeInTheDocument();
  });

  test('Should render H2 Left-Side', () => {
    render(
      <MemoryRouter>
        <EnterOrRegister enterOrRegister={true} />
      </MemoryRouter>
    );

    const headingLeft = screen.getByRole('heading', { name: /Ainda não é cliente Ingresso.com?/i });
    expect(headingLeft).toBeInTheDocument();
  });

  test('Should render H2 Right-Side', () => {
    render(
      <MemoryRouter>
        <EnterOrRegister enterOrRegister={true} />
      </MemoryRouter>
    );

    const headingRight = screen.getByRole('heading', { name: 'Cliente Ingresso.com?' });
    expect(headingRight).toBeInTheDocument();
  });

  test('Should render p', () => {
    render(
      <MemoryRouter>
        <EnterOrRegister enterOrRegister={true} />
      </MemoryRouter>
    );

    const paragraph = screen.getByText(
      'Compre ingressos e combos de pipoca para ir ao cinema com segurança e mais comodidade!'
    );
    expect(paragraph).toBeInTheDocument();
  });

  test('Should render button', () => {
    render(
      <MemoryRouter>
        <EnterOrRegister enterOrRegister={true} />
      </MemoryRouter>
    );

    const button = screen.getByRole('button', { name: 'CRIAR UMA NOVA CONTA' });
    expect(button).toBeInTheDocument();
  });

  test('Should render Tag "a"', () => {
    render(
      <MemoryRouter>
        <EnterOrRegister enterOrRegister={true} />
      </MemoryRouter>
    );

    const linkA = screen.getByText('Entrar na minha Conta');
    expect(linkA).toBeInTheDocument();
  });

  test('should matchSnapshot', () => {
    const { container } = render(
      <MemoryRouter>
        <EnterOrRegister enterOrRegister={true} />
      </MemoryRouter>
    );

    expect(container).toMatchSnapshot();
  });
});
