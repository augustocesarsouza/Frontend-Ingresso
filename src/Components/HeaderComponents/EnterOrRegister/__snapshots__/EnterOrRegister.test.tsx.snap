// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EnterOrRegister should matchSnapshot 1`] = `
.c0 {
  position: absolute;
  left: -254px;
  top: 50px;
  background-color: #191919;
  width: 51rem;
  height: 26rem;
  display: flex;
  padding: 30px 10px;
}

.c0 svg {
  position: relative;
  color: #191919;
  font-size: 39px;
  left: 249px;
  bottom: 52px;
}

.c1 {
  width: 50%;
  padding-right: 15px;
}

.c2 {
  display: flex;
  flex-direction: column;
  gap: 30px;
}

.c3 {
  font-size: 17px;
  color: white;
  font-weight: 600;
}

.c4 {
  font-size: 14px;
  color: white;
}

.c5 {
  padding: 10px;
  border: none;
  background: #0099ff;
  color: #000000;
  font-size: 14px;
  font-weight: 700;
  border-radius: 6px;
  cursor: pointer;
}

.c6 {
  border-right: 1px solid #ffffff5e;
  margin-left: 13px;
}

.c7 {
  width: 50%;
  padding-left: 20px;
  display: flex;
  flex-direction: column;
  gap: 25px;
}

.c8 {
  color: #2196f3;
  font-size: 15px;
  cursor: pointer;
}

<div>
  <div
    class="c0"
  >
    <svg
      aria-hidden="true"
      class="svg-inline--fa fa-caret-up "
      data-icon="caret-up"
      data-prefix="fas"
      focusable="false"
      role="img"
      viewBox="0 0 320 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M182.6 137.4c-12.5-12.5-32.8-12.5-45.3 0l-128 128c-9.2 9.2-11.9 22.9-6.9 34.9s16.6 19.8 29.6 19.8H288c12.9 0 24.6-7.8 29.6-19.8s2.2-25.7-6.9-34.9l-128-128z"
        fill="currentColor"
      />
    </svg>
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <h2
          class="c3"
        >
          Ainda não é cliente Ingresso.com?
        </h2>
        <p
          class="c4"
        >
          Compre ingressos e combos de pipoca para ir ao cinema com segurança e mais comodidade!
        </p>
        <button
          class="c5"
        >
          CRIAR UMA NOVA CONTA
        </button>
      </div>
    </div>
    <div
      class="c6"
    />
    <div
      class="c7"
    >
      <h2
        class="c3"
      >
        Cliente Ingresso.com?
      </h2>
      <a
        class="c8"
      >
        Entrar na minha Conta
      </a>
    </div>
  </div>
</div>
`;
